core/commands/p2p.go

```diff
@@ -53,49 +56,115 @@ are refined`,
 	},
 
 	Subcommands: map[string]*cmds.Command{
-		"listener": p2pListenerCmd,
-		"stream":   p2pStreamCmd,
+		"stream": p2pStreamCmd,
+
+		"forward": p2pForwardCmd,
+		"close":   p2pCloseCmd,
+		"ls":      p2pLsCmd,
 	},
 }
 
-// p2pListenerCmd is the 'ipfs p2p listener' command
-var p2pListenerCmd = &cmds.Command{
+var p2pForwardCmd = &cmds.Command{
 	Helptext: cmdkit.HelpText{
-		Tagline:          "P2P listener management.",
-		ShortDescription: "Create and manage listener p2p endpoints",
+		Tagline: "Forward connections to or from libp2p services",
+		ShortDescription: `
+Forward connections to <listen-address> to <target-address>. Protocol specifies
+the libp2p protocol to use.
+
+To create libp2p service listener, specify '/ipfs' as <listen-address>
+
+Examples:
+  ipfs p2p forward myproto /ipfs /ip4/127.0.0.1/tcp/1234
+    - Forward connections to 'myproto' libp2p service to 127.0.0.1:1234
+
+  ipfs p2p forward myproto /ip4/127.0.0.1/tcp/4567 /ipfs/QmPeer
+    - Forward connections to 127.0.0.1:4567 to 'myproto' service on /ipfs/QmPeer
+
+`,
 	},
+	Arguments: []cmdkit.Argument{
+		cmdkit.StringArg("protocol", true, false, "Protocol identifier."),
+		cmdkit.StringArg("listen-address", true, false, "Listening endpoint"),
+		cmdkit.StringArg("target-address", true, false, "Target endpoint."),
+	},
+	Run: func(req cmds.Request, res cmds.Response) {
+		n, err := p2pGetNode(req)
+		if err != nil {
+			res.SetError(err, cmdkit.ErrNormal)
+			return
+		}
 
-	Subcommands: map[string]*cmds.Command{
-		"ls":    p2pListenerLsCmd,
-		"open":  p2pListenerListenCmd,
-		"close": p2pListenerCloseCmd,
+		//TODO: Do we really want/need implicit prefix?
```

Comment:
Thoughts on this?
Would it cause any restrictions or confusion, with or without them?

Reply:
It basically restricts one from communicating with libp2p services not in `/p2p` namespace (so e.g. you can't use this to talk to other peers' bitswap which uses `/ipfs/bitswap/1.1.0` as protocol name). The restriction was there before (probably put there by me), but now I'm not sure if we really want/need it.

Reply:
It probably makes sense to make the user always type /p2p/foo even if we restrict them to that namespace. That way, in the future, if we remove that restriction, we don't have to do anything weird.